test/algorithm/contains_subrange.cpp

```diff
@@ -0,0 +1,95 @@
+// Range v3 library
+//
+//  Copyright Johel Guerrero 2019
+//  Copyright Google LLC 2021
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#include <array>
+#include <initializer_list>
+
+#include <range/v3/algorithm/contains_subrange.hpp>
+#include <range/v3/iterator/operations.hpp>
+#include <range/v3/view/subrange.hpp>
+
+#include "../simple_test.hpp"
+#include "../test_iterators.hpp"
+
+int comparison_count = 0;
+
+template<typename T>
+bool counting_equals(const T & a, const T & b)
+{
+    ++comparison_count;
+    return a == b;
+}
+
+int main()
+{
+    using namespace ranges;
+    auto full_range = std::array<int, 10>{0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+    auto valid_subrange = std::array<int, 5>{2, 3, 4, 5, 6};
+    auto invalid_subrange = std::array<int, 5>{3, 4, 5, 6, 2};
```

Comment:
With MSVC, `std::array<int, N>::iterator` isn't `int*`, so this test is failing. Can you change these to an `int[N]` instead? The clang 5 build is failing for reasons that are also basically related to this.